### ISSUE TITLE
hyprctl dpms off leads to hyprland crash when OBS open with pipewire screen capture

### ISSUE THREAD

**Issue by jmcelroy01** [hyprwm/xdg-desktop-portal-hyprland#62]
I was able to reproduce this issue 3 times and narrow the cause down to this: 

In hyprland.conf, I have this line - 

`exec-once = swayidle timeout 240 'brightnessctl -s && brightnessctl set 5%' resume 'brightnessctl -r' timeout 300 'swaylock  --screenshots --clock --indicator --indicator-radius 100 --indicator-thickness 7 --effect-blur 7x5 --effect-vignette 0.5:0.5 --key-hl-color 880033 --line-color 00000000 --inside-color 00000088 --separator-color 00000000 --grace 2 --fade-in 0.2' timeout 360 'hyprctl dispatch dpms off' resume 'hyprctl dispatch dpms on'`

If I leave OBS open with a pipewire screen capture (not actively recording), and allow the laptop to dim/lock/turn display off, Hyprland crashed each time immediately after the display turns off. 
This line seemed to occur all 3 times in the logs -
`A backend call failed: Message recipient disconnected from message bus without replying`,
so it looks like `hyprctl dispatch dpms off` turns off the portal somehow, which cascades into Hyprland crashing after xdg-desktop-portal-hyprland and OBS go down. 

There are also a fair number of 
`xdg-desktop-portal-hyprland	  [ERROR] - wayland: failed to connect to display`
lines in the logs. I noticed some inconsistency in OBS prompting for which monitor to capture. I found deleting the pipewire display capture and re-adding it sometimes made it ask which monitor to capture, other times I just opened the settings of the existing pipewire display capture and clicked select monitor a few times and it eventually popped up. I only had a laptop display throughout all of this. This is likely a separate issue, I can open later one if needed. 

--------------------------------
some fastfetch info:

OS: Gentoo 2.13 x86_64
Host: ROG Zephyrus G14 GA402RJ_GA402RJ (1.0)
Kernel: 6.1.28-gentoo-x86_64
Display: 2560x1600 @ 120Hz
Brightness (amdgpu_bl1): 31%
WM: Hyprland (Wayland)
CPU: AMD Ryzen 9 6900HS (16) @ 4.933886 GHz
GPU 1: AMD Radeon Graphics (RADV REMBRANDT)
GPU 2: AMD Radeon RX 6700S (RADV NAVI23) (7.98 GiB)
Vulkan: 1.3.246 - radv [Mesa 23.1.1]
OpenGL: 4.6 (Compatibility Profile) Mesa 23.1.1
OpenCL: 2.0

--------------------------------

Hyprland: 0.25
xdg-desktop-portal-hyprland: 9999 (wayland-desktop overlay)
I have xdg-desktop-portal-gtk installed, no -kde or -gnome. 

--------------------------------

<details>
<summary>Here are the crash logs</summary>

heh, it was too long to paste directly - 

[hyprland-crash-log-obs-xdg-attachment.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11601475/hyprland-crash-log-obs-xdg-attachment.txt)

</details>

Lmk if I can provide any other useful info, and thanks. 

Once 0.26 lands in the Gentoo repo I'll see if it still occurs.

**Comment by jmcelroy01** [comment]
Was able to reproduce with 0.26.

<details>
<summary>Here are the crash logs</summary>

```
	systemd	Started systemd-coredump@2-53382-0.service.
	systemd-coredump	Process 53372 (xdg-desktop-por) of user 1000 dumped core.

Module libicudata.so.73 without build-id.
Module libicuuc.so.73 without build-id.
Module libgmp.so.10 without build-id.
Module libxml2.so.2 without build-id.
Module libtinfo.so.6 without build-id.
Module libz3.so.4.12 without build-id.
Module libgcc_s.so.1 without build-id.
Module libstdc++.so.6 without build-id.
Module libxcb-dri3.so.0 without build-id.
Module libdrm_amdgpu.so.1 without build-id.
Module libdrm_radeon.so.1 without build-id.
Module libsensors.so.5 without build-id.
Module libz.so.1 without build-id.
Module libLLVM-16.so without build-id.
Module libglapi.so.0 without build-id.
Module libpipewire-module-session-manager.so without build-id.
Module libpipewire-module-metadata.so without build-id.
Module libpipewire-module-adapter.so without build-id.
Module libpipewire-module-client-device.so without build-id.
Module libpipewire-module-client-node.so without build-id.
Module libpipewire-module-protocol-native.so without build-id.
Module libdbus-1.so.3 without build-id.
Module libspa-dbus.so without build-id.
Module libspa-journal.so without build-id.
Module libspa-support.so without build-id.
Module libXdmcp.so.6 without build-id.
Module libXau.so.6 without build-id.
Module libxcb.so.1 without build-id.
Module libgpg-error.so.0 without build-id.
Module libm.so.6 without build-id.
Module libxcb-randr.so.0 without build-id.
Module libexpat.so.1 without build-id.
Module libwayland-server.so.0 without build-id.
Module ld-linux-x86-64.so.2 without build-id.
Module liblz4.so.1 without build-id.
Module libzstd.so.1 without build-id.
Module liblzma.so.5 without build-id.
Module libgcrypt.so.20 without build-id.
Module libcap.so.2 without build-id.
Module libffi.so.8 without build-id.
Module libc.so.6 without build-id.
Module libuuid.so.1 without build-id.
Module libdrm.so.2 without build-id.
Module libgbm.so.1 without build-id.
Module libinih.so.0 without build-id.
Module libpipewire-0.3.so.0 without build-id.
Module libsystemd.so.0 without build-id.
Module libwayland-client.so.0 without build-id.
Module xdg-desktop-portal-hyprland without build-id.
Stack trace of thread 53372:
#0  0x00007f96feaaa56b wl_list_remove (libwayland-client.so.0 + 0xb56b)
#1  0x00005605f9f2003b n/a (xdg-desktop-portal-hyprland + 0xb03b)
#2  0x00007f96fe9de555 n/a (libsystemd.so.0 + 0x39555)
#3  0x00007f96fe9dfb7b n/a (libsystemd.so.0 + 0x3ab7b)
#4  0x00007f96fe9f2fc5 n/a (libsystemd.so.0 + 0x4dfc5)
#5  0x00005605f9f1c8a1 n/a (xdg-desktop-portal-hyprland + 0x78a1)
#6  0x00007f96fe6b0650 n/a (libc.so.6 + 0x27650)
#7  0x00007f96fe6b0709 __libc_start_main (libc.so.6 + 0x27709)
#8  0x00005605f9f1ca95 n/a (xdg-desktop-portal-hyprland + 0x7a95)

Stack trace of thread 53374:
#0  0x00007f96fe712a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007f96fe715352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007f96fc4d7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007f96fc48859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007f96fc4d7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007f96fe715feb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007f96fe799b50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53373:
#0  0x00007f96fe79903a epoll_wait (libc.so.6 + 0x11003a)
#1  0x00007f96feaeba68 n/a (libspa-support.so + 0x19a68)
#2  0x00007f96feadc395 n/a (libspa-support.so + 0xa395)
#3  0x00007f96fe8f0490 n/a (libpipewire-0.3.so.0 + 0x4c490)
#4  0x00007f96fe715feb n/a (libc.so.6 + 0x8cfeb)
#5  0x00007f96fe799b50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53376:
#0  0x00007f96fe712a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007f96fe715352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007f96fc4d7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007f96fc48859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007f96fc4d7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007f96fe715feb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007f96fe799b50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53377:
#0  0x00007f96fe712a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007f96fe715352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007f96fc4d7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007f96fc48859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007f96fc4d7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007f96fe715feb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007f96fe799b50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53375:
#0  0x00007f96fe712a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007f96fe715352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007f96fc4d7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007f96fc48859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007f96fc4d7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007f96fe715feb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007f96fe799b50 n/a (libc.so.6 + 0x110b50)
ELF object binary architecture: AMD x86-64

	systemd	systemd-coredump@2-53382-0.service: Deactivated successfully.
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=dumped, status=11/SEGV
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'core-dump'.
	dbus-daemon	[session uid=1000 pid=4408] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.hyprland' unit='xdg-desktop-portal-hyprland.service' requested by ':1.13' (uid=1000 pid=4659 comm="/usr/libexec/xdg-desktop-portal" label="kernel")
	user@1000.service	A backend call failed: Message recipient disconnected from message bus without replying
	user@1000.service	Error deleting permission: No entry for 2d3f5c7a-dc4b-49e4-b2b1-b1f9f6f192c5
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 1.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	Starting xdg-desktop-portal-hyprland.service...
	dbus-daemon	[session uid=1000 pid=4408] Successfully activated service 'org.freedesktop.impl.portal.desktop.hyprland'
	systemd	Started xdg-desktop-portal-hyprland.service.
	user@1000.service	Failed to close session implementation: Timeout was reached
	clamd	Mon Jun  5 07:03:41 2023 -> SelfCheck: Database status OK.
	user@1000.service	<WpPortalPermissionStorePlugin:0x55a17c3e1140> Failed to call Lookup: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for camera
	systemd	Started systemd-coredump@3-54585-0.service.
	systemd-coredump	Process 52795 (Hyprland) of user 1000 dumped core.

Module libxcb-dri2.so.0 without build-id.
Module libEGL_mesa.so.0 without build-id.
Module libdrm_radeon.so.1 without build-id.
Module libsensors.so.5 without build-id.
Module libglapi.so.0 without build-id.
Module libcrypto.so.3 without build-id.
Module libssl.so without build-id.
Module libVkLayer_MESA_device_select.so without build-id.
Module libicudata.so.73 without build-id.
Module libicuuc.so.73 without build-id.
Module libgmp.so.10 without build-id.
Module libxml2.so.2 without build-id.
Module libtinfo.so.6 without build-id.
Module libz3.so.4.12 without build-id.
Module libxshmfence.so.1 without build-id.
Module libxcb-sync.so.1 without build-id.
Module libX11-xcb.so.1 without build-id.
Module libdrm_amdgpu.so.1 without build-id.
Module libLLVM-16.so without build-id.
Module libblkid.so.1 without build-id.
Module libbrotlicommon.so.1 without build-id.
Module libgraphite2.so.3 without build-id.
Module libmount.so.1 without build-id.
Module libgmodule-2.0.so.0 without build-id.
Module libpcre2-8.so.0 without build-id.
Module libgpg-error.so.0 without build-id.
Module libbrotlidec.so.1 without build-id.
Module libbz2.so.1 without build-id.
Module libpangoft2-1.0.so.0 without build-id.
Module libharfbuzz.so.0 without build-id.
Module libfribidi.so.0 without build-id.
Module libgio-2.0.so.0 without build-id.
Module libglib-2.0.so.0 without build-id.
Module liblz4.so.1 without build-id.
Module libzstd.so.1 without build-id.
Module liblzma.so.5 without build-id.
Module libgcrypt.so.20 without build-id.
Module libXrender.so.1 without build-id.
Module libXext.so.6 without build-id.
Module libX11.so.6 without build-id.
Module libfreetype.so.6 without build-id.
Module libfontconfig.so.1 without build-id.
Module libpng16.so.16 without build-id.
Module libz.so.1 without build-id.
Module libXdmcp.so.6 without build-id.
Module libXau.so.6 without build-id.
Module libevdev.so.2 without build-id.
Module libmtdev.so.1 without build-id.
Module libcap.so.2 without build-id.
Module libxcb-randr.so.0 without build-id.
Module libexpat.so.1 without build-id.
Module libGLdispatch.so.0 without build-id.
Module ld-linux-x86-64.so.2 without build-id.
Module libffi.so.8 without build-id.
Module libc.so.6 without build-id.
Module libgcc_s.so.1 without build-id.
Module libm.so.6 without build-id.
Module libstdc++.so.6 without build-id.
Module libpangocairo-1.0.so.0 without build-id.
Module libgobject-2.0.so.0 without build-id.
Module libpango-1.0.so.0 without build-id.
Module libsystemd.so.0 without build-id.
Module libcairo.so.2 without build-id.
Module libxcb-res.so.0 without build-id.
Module libxcb-icccm.so.4 without build-id.
Module libxcb-ewmh.so.2 without build-id.
Module libxcb-composite.so.0 without build-id.
Module libxcb-xinput.so.0 without build-id.
Module libxcb-xfixes.so.0 without build-id.
Module libxcb-shm.so.0 without build-id.
Module libxcb-render-util.so.0 without build-id.
Module libxcb-render.so.0 without build-id.
Module libxcb-present.so.0 without build-id.
Module libxcb-dri3.so.0 without build-id.
Module libxcb.so.1 without build-id.
Module libinput.so.10 without build-id.
Module libliftoff.so.0 without build-id.
Module libdisplay-info.so.1 without build-id.
Module libseat.so.1 without build-id.
Module libudev.so.1 without build-id.
Module libvulkan.so.1 without build-id.
Module libGLESv2.so.2 without build-id.
Module libgbm.so.1 without build-id.
Module libEGL.so.1 without build-id.
Module libpixman-1.so.0 without build-id.
Module libxkbcommon.so.0 without build-id.
Module libdrm.so.2 without build-id.
Module libwayland-client.so.0 without build-id.
Module libwayland-server.so.0 without build-id.
Module Hyprland without build-id.
Stack trace of thread 52795:
#0  0x00007fcbb2fcdd4c n/a (libc.so.6 + 0x8ed4c)
#1  0x00007fcbb2f7bd96 raise (libc.so.6 + 0x3cd96)
#2  0x00007fcbb2f657fc abort (libc.so.6 + 0x267fc)
#3  0x000055fb31f19515 _Z25handleUnrecoverableSignali (Hyprland + 0x10f515)
#4  0x00007fcbb2f7be40 n/a (libc.so.6 + 0x3ce40)
#5  0x000055fb3202db34 wlr_buffer_get_dmabuf (Hyprland + 0x223b34)
#6  0x000055fb31ff3af6 gles2_texture_from_buffer (Hyprland + 0x1e9af6)
#7  0x000055fb31effc03 _ZN26CScreencopyProtocolManager15copyFrameDmabufEP16SScreencopyFrame (Hyprland + 0xf5c03)
#8  0x000055fb31effdbb _ZN26CScreencopyProtocolManager10shareFrameEP16SScreencopyFrame (Hyprland + 0xf5dbb)
#9  0x000055fb31f017c2 _ZN26CScreencopyProtocolManager14shareAllFramesEP8CMonitorb (Hyprland + 0xf77c2)
#10 0x000055fb31f01bca _ZN26CScreencopyProtocolManager14onOutputCommitEP8CMonitorP23wlr_output_event_commit (Hyprland + 0xf7bca)
#11 0x000055fb31f84160 _Z13handleWrappedP11wl_listenerPv (Hyprland + 0x17a160)
#12 0x00007fcbb3c02b3c wl_signal_emit_mutable (libwayland-server.so.0 + 0x9b3c)
#13 0x000055fb3201d65a wlr_output_commit_state (Hyprland + 0x21365a)
#14 0x000055fb3201d94d wlr_output_commit (Hyprland + 0x21394d)
#15 0x000055fb31faa13e _ZN15CKeybindManager4dpmsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE (Hyprland + 0x1a013e)
#16 0x000055fb31fb3193 _ZNSt17_Function_handlerIFvNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEPS6_E9_M_invokeERKSt9_Any_dataOS5_ (Hyprland + 0x1a9193)
#17 0x000055fb31f0b12c _Z15dispatchRequestNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE (Hyprland + 0x10112c)
#18 0x000055fb31f13390 _Z8getReplyNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE (Hyprland + 0x109390)
#19 0x000055fb31f140a4 _Z13hyprCtlFDTickijPv (Hyprland + 0x10a0a4)
#20 0x00007fcbb3c04b32 wl_event_loop_dispatch (libwayland-server.so.0 + 0xbb32)
#21 0x00007fcbb3c02575 wl_display_run (libwayland-server.so.0 + 0x9575)
#22 0x000055fb31eb9cfa main (Hyprland + 0xafcfa)
#23 0x00007fcbb2f66650 n/a (libc.so.6 + 0x27650)
#24 0x00007fcbb2f66709 __libc_start_main (libc.so.6 + 0x27709)
#25 0x000055fb31ec7465 _start (Hyprland + 0xbd465)

Stack trace of thread 52808:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52814:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52806:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52810:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52801:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcbaea2329d n/a (libvulkan_radeon.so + 0x22329d)
#3  0x00007fcbae9f453b n/a (libvulkan_radeon.so + 0x1f453b)
#4  0x00007fcbaea231cb n/a (libvulkan_radeon.so + 0x2231cb)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52812:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52802:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcbaea2329d n/a (libvulkan_radeon.so + 0x22329d)
#3  0x00007fcbae9f453b n/a (libvulkan_radeon.so + 0x1f453b)
#4  0x00007fcbaea231cb n/a (libvulkan_radeon.so + 0x2231cb)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52804:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52811:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52820:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52803:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52815:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52818:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52805:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53133:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53135:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52809:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52813:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52807:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52816:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 53134:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52819:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52817:
#0  0x00007fcbb2fc8a45 n/a (libc.so.6 + 0x89a45)
#1  0x00007fcbb2fcb352 pthread_cond_wait (libc.so.6 + 0x8c352)
#2  0x00007fcb9acd7f2d n/a (radeonsi_dri.so + 0xd7f2d)
#3  0x00007fcb9ac8859b n/a (radeonsi_dri.so + 0x8859b)
#4  0x00007fcb9acd7e5b n/a (radeonsi_dri.so + 0xd7e5b)
#5  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#6  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)

Stack trace of thread 52821:
#0  0x00007fcbb305097f accept (libc.so.6 + 0x11197f)
#1  0x000055fb31fd2acd n/a (Hyprland + 0x1c8acd)
#2  0x00007fcbb3308453 n/a (libstdc++.so.6 + 0x108453)
#3  0x00007fcbb2fcbfeb n/a (libc.so.6 + 0x8cfeb)
#4  0x00007fcbb304fb50 n/a (libc.so.6 + 0x110b50)
ELF object binary architecture: AMD x86-64

	yambar	foreign-toplevel: disconnected from the Wayland compositor
	yambar	bar:wayland: disconnected from wayland
		Error reading events from display: Broken pipe
	yambar	bar: module: CENTER #0 (toplevel): non-zero exit value: -1
	systemd	xdg-desktop-portal-gtk.service: Main process exited, code=exited, status=1/FAILURE
	polkitd	Unregistered Authentication Agent for unix-session:1 (system bus name :1.92, object path /org/kde/PolicyKit1/AuthenticationAgent, locale en_US.utf8) (disconnected from bus)
	systemd	xdg-desktop-portal-gtk.service: Failed with result 'exit-code'.
	systemd	xdg-desktop-portal-hyprland.service: Consumed 7.388s CPU time.
	dbus-daemon	[session uid=1000 pid=4408] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.hyprland' unit='xdg-desktop-portal-hyprland.service' requested by ':1.13' (uid=1000 pid=4659 comm="/usr/libexec/xdg-desktop-portal" label="kernel")
	systemd	Starting xdg-desktop-portal-hyprland.service...
	xdg-desktop-portal-hyprland	2023/06/05 07:09:48 [ERROR] - wayland: failed to connect to display
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=exited, status=1/FAILURE
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 1.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	Starting xdg-desktop-portal-hyprland.service...
	xdg-desktop-portal-hyprland	2023/06/05 07:09:48 [ERROR] - wayland: failed to connect to display
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=exited, status=1/FAILURE
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 2.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	Starting xdg-desktop-portal-hyprland.service...
	xdg-desktop-portal-hyprland	2023/06/05 07:09:48 [ERROR] - wayland: failed to connect to display
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=exited, status=1/FAILURE
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 3.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	Starting xdg-desktop-portal-hyprland.service...
	xdg-desktop-portal-hyprland	2023/06/05 07:09:49 [ERROR] - wayland: failed to connect to display
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=exited, status=1/FAILURE
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
	systemd	systemd-coredump@3-54585-0.service: Deactivated successfully.
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 4.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	Starting xdg-desktop-portal-hyprland.service...
	xdg-desktop-portal-hyprland	2023/06/05 07:09:49 [ERROR] - wayland: failed to connect to display
	systemd	xdg-desktop-portal-hyprland.service: Main process exited, code=exited, status=1/FAILURE
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
	systemd	xdg-desktop-portal-hyprland.service: Scheduled restart job, restart counter is at 5.
	systemd	Stopped xdg-desktop-portal-hyprland.service.
	systemd	xdg-desktop-portal-hyprland.service: Start request repeated too quickly.
	systemd	xdg-desktop-portal-hyprland.service: Failed with result 'exit-code'.
	systemd	Failed to start xdg-desktop-portal-hyprland.service.
```

</details>

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11715646/patch.txt)

try this patch

**Comment by jmcelroy01** [comment]
After patching/testing, what is the command to uninstall a manually installed xdg-desktop-portal-hyprland so the normal one can be reinstalled from portage without file collision issues?

**Comment by jmcelroy01** [comment]
Just cloned the repo and built. There is no /src/protocols/Screencopy.cpp file to patch?

**Comment by jmcelroy01** [comment]
Finally learned about Gentoo [patching](https://wiki.gentoo.org/wiki//etc/portage/patches). So I tried the file using that approach and it also complains about missing Screencopy.cpp. 

<details>
<summary>Build log</summary>

```
 * Package:    gui-libs/xdg-desktop-portal-hyprland-0.4.0:0/9999
 * Repository: guru
 * Maintainer: remigiusz.micielski@gmail.com
 * Upstream:   https://github.com/hyprwm/xdg-desktop-portal-hyprland/issues
 * USE:        abi_x86_64 amd64 elibc_glibc kernel_linux systemd
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
>>> Unpacking source...
>>> Unpacking xdg-desktop-hyprland-0.4.0.tar.gz to /var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/work
>>> Source unpacked in /var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/work
>>> Preparing source in /var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/work/xdg-desktop-portal-hyprland-0.4.0 ...
 * ===========================================================================================================================
 * Applying user patches from /etc/portage/patches ...
 * Applying obs.patch ...
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/src/protocols/Screencopy.cpp b/src/protocols/Screencopy.cpp
|index d47888a0..7867d510 100644
|--- a/src/protocols/Screencopy.cpp
|+++ b/src/protocols/Screencopy.cpp
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 19
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/subprojects/wlroots b/subprojects/wlroots
|--- a/subprojects/wlroots
|+++ b/subprojects/wlroots
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored                                                                                                [ !! ]
 * ERROR: gui-libs/xdg-desktop-portal-hyprland-0.4.0::guru failed (prepare phase):
 *   patch -p1  failed with /etc/portage/patches/gui-libs/xdg-desktop-portal-hyprland-0.4.0/obs.patch
 *
 * Call stack:
 *               ebuild.sh, line  136:  Called src_prepare
 *             environment, line 2288:  Called default
 *      phase-functions.sh, line  872:  Called default_src_prepare
 *      phase-functions.sh, line  948:  Called __eapi8_src_prepare
 *             environment, line  385:  Called eapply_user
 *             environment, line 1124:  Called eapply '/etc/portage/patches/gui-libs/xdg-desktop-portal-hyprland-0.4.0/obs.patch'
 *             environment, line 1084:  Called _eapply_patch '/etc/portage/patches/gui-libs/xdg-desktop-portal-hyprland-0.4.0/obs.patch'
 *             environment, line 1022:  Called __helpers_die 'patch -p1  failed with /etc/portage/patches/gui-libs/xdg-desktop-portal-hyprland-0.4.0/obs.patch'
 *   isolated-functions.sh, line  112:  Called die
 * The specific snippet of code:
 *   		die "$@"
 *
 * If you need support, post the output of `emerge --info '=gui-libs/xdg-desktop-portal-hyprland-0.4.0::guru'`,
 * the complete build log and the output of `emerge -pqv '=gui-libs/xdg-desktop-portal-hyprland-0.4.0::guru'`.
 * The complete build log is located at '/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/temp/environment'.
 * Working directory: '/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/work/xdg-desktop-portal-hyprland-0.4.0'
 * S: '/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-0.4.0/work/xdg-desktop-portal-hyprland-0.4.0'
```
</details>

**Comment by vaxerski** [comment]
closing as if anything this is a hyprland issue not xdph

**Comment by VDFOREVER** [comment]
Same problem, but on artix
